Python/specialize.c

```diff
@@ -362,94 +427,134 @@ _Py_Specialize_LoadAttr(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name, Sp
         goto success;
     }
     PyTypeObject *type = Py_TYPE(owner);
-    if (type->tp_getattro != PyObject_GenericGetAttr) {
-        SPECIALIZATION_FAIL(LOAD_ATTR, Py_TYPE(owner), name, "__getattribute__ overridden");
-        goto fail;
-    }
     if (type->tp_dict == NULL) {
         if (PyType_Ready(type) < 0) {
             return -1;
         }
     }
-    PyObject *descr = _PyType_Lookup(type, name);
-    if (descr != NULL) {
-        // We found an attribute with a data-like descriptor.
-        PyTypeObject *dtype = Py_TYPE(descr);
-        if (dtype != &PyMemberDescr_Type) {
-            SPECIALIZATION_FAIL(LOAD_ATTR, Py_TYPE(owner), name, "not a member descriptor");
+    PyObject *descr;
+    DesciptorClassification kind = analyze_descriptor(type, name, &descr);
+    switch(kind) {
+        case OVERRIDING:
+            SPECIALIZATION_FAIL(LOAD_ATTR, type, name, "overriding descriptor");
             goto fail;
-        }
-        // It's a slot
-        PyMemberDescrObject *member = (PyMemberDescrObject *)descr;
-        struct PyMemberDef *dmem = member->d_member;
-        if (dmem->type != T_OBJECT_EX) {
-            SPECIALIZATION_FAIL(LOAD_ATTR, Py_TYPE(owner), name, "non-object slot");
+        case METHOD:
+            SPECIALIZATION_FAIL(LOAD_ATTR, type, name, "method");
```

Comment:
Maybe?
There is nothing preventing us from specializing it, but I don't know how common this case is.

When we specialize `LOAD_METHOD` it will probably be top of the list.